### PR TITLE
Fix classification labels for batch size 1

### PR DESCRIPTION
**Pull request by lukasugar**
When using `batch_size=1` for smoke tests, `squeeze()` turns a [1, 1] label tensor into a scalar for batch_size=1,
which breaks CrossEntropyLoss with a batch-size mismatch. `view(-1)`keeps labels shaped as [B].

For batch sizes other than 1, the behavior remains the same.

### CODE REVIEW

**Comment by Sllambias** [comment]
Seems to also be included in #6 - just porting my comment here:

I am concerned this may cause issues for classification/regression inference. Have you tested that?
(And this shows that the full pipeline tests are long overdue..)

**Comment by lukasugar** [comment]
> Seems to also be included in #6 - just porting my comment here:
> 
> I am concerned this may cause issues for classification/regression inference. Have you tested that? (And this shows that the full pipeline tests are long overdue..)

Replied there:
> I've tested both classification and regression and it works.
`squeeze()` and `.view(-1)` should behave the same - with view have support for scalars - for the current evals/finetuning setup.

Maybe we can close this PR and look at all the changes in the other one
